elasticsearch-vmss/azuredeploy.json

```diff
@@ -0,0 +1,504 @@
+        "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version."
+      }
+    },
+    "vmssName": {
+      "type": "string",
+      "metadata": {
+        "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure. A hash is prepended to this string for some resources, and resource-specific information is appended."
+      },
+      "maxLength": 61
+    },
+    "instanceCount": {
+      "type": "int",
+      "metadata": {
+        "description": "Number of VM instances (1000 or less)."
+      },
+      "maxValue": 1000
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username on all VMs."
+      }
+    },
+    "adminKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Public key for ssh access"
+      }
+    },
+    "templateBase": {
+      "type": "string",
+      "defaultValue": "https://raw.githubusercontent.com/hglkrijger/azure-quickstart-templates/master",
+      "metadata": {
+        "description": "Change this value to your repo name if deploying from a fork"
+      }
+    }
+  },
+  "variables": {
+    "storageAccountType": "Standard_LRS",
+    "namingInfix": "[toLower(parameters('vmssName'))]",
+    "longNamingInfix": "[resourceGroup().name]",
+    "domainNameLabel": "[variables('namingInfix')]",
+    "addressPrefix": "10.0.0.0/8",
+    "subnetPrefix": "10.0.0.0/16",
+    "virtualNetworkName": "[concat(variables('namingInfix'), '-vnet')]",
+    "publicIpAddressName": "[concat(variables('namingInfix'), '-pip')]",
+    "publicIpAddressId": "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIpAddressName'))]",
+    "subnetName": "[concat(variables('namingInfix'), '-subnet')]",
+    "loadBalancerName": "[concat(variables('namingInfix'), '-lb')]",
+    "loadBalancerId": "[resourceId('Microsoft.Network/loadBalancers', variables('loadBalancerName'))]",
+    "backendPoolName": "[concat(variables('namingInfix'), '-bepool')]",
+    "nicName": "[concat(variables('namingInfix'), '-nic')]",
+    "masterName": "[concat(variables('namingInfix'), '-master')]",
+    "availabilitySetName": "[concat(variables('namingInfix'), '-master-set')]",
+    "nsgName": "[concat(variables('namingInfix'), '-master-nsg')]",
+    "ipConfigName": "[concat(variables('namingInfix'), '-ipconfig')]",
+    "frontendIpConfigId": "[concat(variables('loadBalancerId'),'/frontendIPConfigurations/loadBalancerFrontEnd')]",
+    "loadBalancerPoolId": "[concat(variables('loadBalancerId'),'/backendAddressPools/', variables('backendPoolName'))]",
+    "loadBalancerProbeId": "[concat(variables('loadBalancerId'),'/probes/tcpProbe')]",
+    "sshProbeId": "[concat(variables('loadBalancerId'),'/probes/sshProbe')]",
+    "masterStorageAccountName": "[concat(uniqueString(resourceGroup().id, variables('namingInfix')), 'master')]",
+    "osType": {
```

Comment:
this is only used once (in the variable below) so can remove this and just put the value in imageReference